tmail-backend/webadmin/webadmin-mailbox/src/main/java/com/linagora/tmail/webadmin/cleanup/CleanupTrashService.java

```diff
@@ -32,20 +39,26 @@
 
 public class CleanupTrashService {
     private static final Logger LOGGER = LoggerFactory.getLogger(CleanupTrashService.class);
-    public static final String WEEKLY_PERIOD = "weekly";
-    public static final String MONTHLY_PERIOD = "monthly";
-    public static final String TRASH_CLEANUP_ENABLED_SETTING = "trash.cleanup.enabled";
-    public static final String TRASH_CLEANUP_PERIOD_SETTING = "trash.cleanup.period";
+    public static final JmapSettingsValue WEEKLY_PERIOD = new JmapSettingsValue("weekly");
+    public static final JmapSettingsValue MONTHLY_PERIOD = new JmapSettingsValue("monthly");
+    public static final JmapSettingsKey TRASH_CLEANUP_ENABLED_SETTING = JmapSettings.JmapSettingsKey$.MODULE$.liftOrThrow("trash.cleanup.enabled");
+    public static final JmapSettingsKey TRASH_CLEANUP_PERIOD_SETTING = JmapSettings.JmapSettingsKey$.MODULE$.liftOrThrow("trash.cleanup.period");
+    private static final Map<JmapSettingsValue, Period> mapPeriodSettingToDuration = ImmutableMap.of(WEEKLY_PERIOD,  Period.ofWeeks(1),
+        MONTHLY_PERIOD, Period.ofMonths(1));
```

Comment:
Should we abstract the setting key and default setting value in `JmapSettings` class?
Something like:
```scala
case class JmapSettings(settings: Map[JmapSettingsKey, JmapSettingsValue], state: UuidState) {
  def trashCleanupEnable(): Boolean =
    settings.get(JmapSettingsKey.liftOrThrow("trash.cleanup.enabled"))
      .map(value => stringToBoolean)
      .getOrElse(false)
}
```